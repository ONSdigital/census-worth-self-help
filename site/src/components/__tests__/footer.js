import React from "react"
import renderer from "react-test-renderer"
import { render, fireEvent } from "react-testing-library"

import Footer from "../footer"

import { navigate } from "gatsby"
jest.mock('gatsby', () => ({
  navigate: jest.fn(),
}))

describe("Footer", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Footer/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("test support page click has analytic event", () => {
  	window._paq = []

  	const { getByText } = render(<Footer phone_link={true}/>);
  	// Census Field Support
  	fireEvent.click(getByText('Census Field Support'));

  	expect(navigate).toHaveBeenCalledTimes(1)
    expect(navigate).toHaveBeenCalledWith('/contactcentre/')

  	expect(window._paq[0]).toEqual([
      "trackEvent",
      "census-field-support",
      "clicked help",
      "/",
      1
    ])
  })
})
