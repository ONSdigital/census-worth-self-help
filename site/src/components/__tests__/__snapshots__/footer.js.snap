// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div
  css={
    Object {
      "map": undefined,
      "name": "1q7gxz7",
      "next": undefined,
      "styles": "
        
    margin: 30px 0px;
  
        width: 100%;
        padding: 10px 0px;
        background-color: ;
        display: flex;
      ",
    }
  }
/>
`;
